pkg/templates/templates.go

```diff
@@ -130,8 +130,8 @@ type TemplateResolver struct {
 }
 
 type TemplateResult struct {
-	resolvedJSON   []byte
-	referencedObjs []client.ObjectIdentifier
+	ResolvedJSON   []byte
+	ReferencedObjs []client.ObjectIdentifier
```

Comment:
Optional: It doesn't add a lot of characters and it improves the readability.
```suggestion
	ReferencedObjects []client.ObjectIdentifier
```